### PR TITLE
Implemented dynamically loading of newly added chatroom

### PR DESCRIPTION
**Pull request by shiBinH**
#### Revision 2
- implemented a check for empty room names
- modified _Start Chatting!_ button so that only only valid room names close the modal

#### Description
- Implemented dynamic reloading of a user's chatroom list after creating a new chatroom.
- Modified Auth service so that existing user data is not erased

#### Testing
- `ng build` and `npm start`
- tested locally that newly added chatrooms are reflected in the ui

### CODE REVIEW

**Comment by FarhanHyder** [comment]
i tested locally. and it works perfectly for dynamically loading the newly added chatroom. great job :D 

suggestion for adding new person to the chatroom:
- if the person already has an account in CC, s/he will be added to that chatroom directly. 
- if the person does not have an account, we can just send an invitation link with the sign in page of CC.

File: src/app/createchannel/createchannel.component.ts [hunk 1]
```diff
@@ -12,32 +11,27 @@ import { UserInfoService } from '../services/user-info.service';
...
-   ownerID: string;
+    roomName: string;
 
-    ngOnInit() {
-   }
+  ngOnInit() {}
 
   sendToFirebase(): void {
     this.chatroomService.addNewChatroom('public', this.roomName, [], this.data.ownerID)
-    .then((result: any) => {
```
Comment (bidachen) [inline src/app/createchannel/createchannel.component.ts]: we might need to check if the user enter a chatroom name before create a new chatroom here?

Comment (shiBinH) [reply src/app/createchannel/createchannel.component.ts]: Good point. For now, I think any non-empty, space trimmed string would do for room name?

**Comment by shiBinH** [comment]
> i tested locally. and it works perfectly for dynamically loading the newly added chatroom. great job :D
> 
> suggestion for adding new person to the chatroom:
> 
> * if the person already has an account in CC, s/he will be added to that chatroom directly.
> * if the person does not have an account, we can just send an invitation link with the sign in page of CC.

The plan for this sprint will be to only adding existing users to chatrooms. I'm not sure if we'll get to invitation links for non-registered users.

**Pull request merged**